apps/smart-contracts/token/test/Vesting.test.ts

```diff
@@ -1054,4 +1129,260 @@ describe('Vesting', () => {
+    const VESTING_PPO_SUPPLY = ONE_PPO.mul(100)
+    let disallowedSourceUser: SignerWithAddress
+    let disallowedDestinationUser: SignerWithAddress
+    let blockedRecipient: SignerWithAddress
+
+    beforeEach(async () => {
+      await setupEndToEndStack()
+      // Mint PPO to Vesting
+      await ppo.connect(owner).mint(vesting.address, VESTING_PPO_SUPPLY)
+      recipients = [
+        unblockedUser1.address,
+        unblockedUser2.address,
+        allowedSourceUser.address,
+        allowedDestinationUser.address,
+        blockedUser.address,
+      ]
+      amountsAllocated = [ONE_PPO, ONE_PPO.mul(2), ONE_PPO.mul(3), ONE_PPO.mul(2), ONE_PPO]
+      vestingStartTime = (await utils.getLastTimestamp(ethers.provider)) + DAY_IN_SECONDS
+      vestingEndTime = vestingStartTime + 3 * YEAR_IN_SECONDS
+      currentTime = await utils.getLastTimestamp(ethers.provider)
+      await vesting.connect(owner).setVestingEndTime(vestingEndTime)
+      await vesting.connect(owner).setVestingStartTime(vestingStartTime)
+      timeAfterVestingStarted = vestingStartTime + BLOCK_DURATION_IN_SECONDS
+      await vesting.connect(owner).setAllocations(recipients, amountsAllocated)
+    })
+
+    it('reverts claim if user blocked', async () => {
+      expect(await blocklist.isIncluded(blockedUser.address)).to.eq(true)
+      expect(await sourceAllowlist.isIncluded(vesting.address)).to.eq(true)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      await mineBlock(ethers.provider as any, timeAfterVestingStarted)
+      expect(await vesting.getClaimableAmount(blockedUser.address)).to.be.gt(0)
+
+      await expect(vesting.connect(blockedUser).claim()).revertedWith('Recipient blocked')
+    })
+
+    it('claims if user not blocked', async () => {
+      expect(await blocklist.isIncluded(unblockedUser1.address)).to.eq(false)
+      const unblockedUserBalanceBefore = await ppo.balanceOf(unblockedUser1.address)
+      const vestingBalanceBefore = await ppo.balanceOf(vesting.address)
+      const unblockedUserTotalClaimedBefore = await vesting.getClaimedAmount(unblockedUser1.address)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      await setNextTimestamp(ethers.provider as any, timeAfterVestingStarted)
+
+      await vesting.connect(unblockedUser1).claim()
+
+      const unblockedUserTotalClaimedAfter = await vesting.getClaimedAmount(unblockedUser1.address)
+      const expectedChangeInBalance = unblockedUserTotalClaimedAfter.sub(
+        unblockedUserTotalClaimedBefore
+      )
+      expect(expectedChangeInBalance).to.not.eq(0)
+      expect(await ppo.balanceOf(unblockedUser1.address)).to.eq(
+        unblockedUserBalanceBefore.add(expectedChangeInBalance)
+      )
+      expect(await ppo.balanceOf(vesting.address)).to.eq(
+        vestingBalanceBefore.sub(expectedChangeInBalance)
+      )
+    })
+
+    describe('if user claims and is blocked after first claim', () => {
```

Comment:
```suggestion
    describe('if user claims, but is subsequently blocked', () => {
```
I think to maintain consistency, lets rename this to match the other describe blocks.